File: filter-pattern/python/src/Person.py
# -*- coding: utf-8 -*-
"""
@author: jarry
"""


# 定义一个实体类，用来过滤的对象
class Person:

    def __init__(self, name, gender, status):
        self.name = name
        self.gender = gender
        self.status = status

    def get_name(self):
        return self.name

    def get_gender(self):
        return self.gender

    def get_status(self):
        return self.status

    def to_string(self):
        return 'Person : [ Name : ' + self.get_name(
        ) + ', Gender : ' + self.get_gender(
        ) + ', Marital Status : ' + self.get_status() + ' ]'

File: filter-pattern/js/src/Person.js
// 定义一个实体类，用来过滤的对象
export class Person {
  constructor(name, gender, status) {
    this.name = name
    this.gender = gender
    this.status = status
  }

  getName() {
    return this.name
  }

  getGender() {
    return this.gender
  }

  getStatus() {
    return this.status
  }

  toString() {
    return 'Person : [ Name : ' +
      this.getName() +
      ', Gender : ' +
      this.getGender() +
      ', Marital Status : ' +
      this.getStatus() +
      ' ]'
  }
}